src/calendaragendamodel.cpp

```diff
@@ -197,6 +197,10 @@ void CalendarAgendaModel::doRefresh(QList<CalendarEventOccurrence *> newEvents)
         // Skip matching events
         while (eventsCounter < events.count() && newEventsCounter < newEvents.count() &&
                eventsEqual(newEvents.at(newEventsCounter), events.at(eventsCounter))) {
+            // We don't know if the event has changed or not,
+            // it may have been reloaded with a different time or summary
+            // because of background sync or timezone change.
+            dataChanged(index(eventsCounter), index(eventsCounter));
```

Comment:
Hm, the eventsEqual() check above gets my first attention. If it's returning true should they be just the same events or with the same content as earlier? Now the check is somewhere in between when it checks the time and uid, but not the other data. Maybe there could be some possibility for better comparison, or maybe separate uid and the other things.

Reply:
For another thing, the model here serves data with three roles, like event occurrence and event pointers. If those values don't change, there shouldn't ideally be dataChanged() signalling.

Could be that the timezone change is a special case and the summary and such are already working? Now the start/end time labels seem to be coming from model.occurrence, and marked CONSTANT. Pondering if simplest just to detect the timezone change and consider all the events changed.

Reply:
Yes, I fully agree. I was planning to expose the time zone change to the manager, so it can emit date time change on manage objects.